src/tbb/misc_ex.cpp

```diff
@@ -312,15 +313,28 @@ static void initialize_hardware_concurrency_info () {
             if ( TBB_GetThreadGroupAffinity( GetCurrentThread(), &ga ) )
                 ProcessorGroupInfo::HoleIndex = ga.Group;
             int nprocs = 0;
+            int min_procs = INT_MAX;
             for ( WORD i = 0; i < ProcessorGroupInfo::NumGroups; ++i ) {
                 ProcessorGroupInfo  &pgi = theProcessorGroups[i];
                 pgi.numProcs = (int)TBB_GetActiveProcessorCount(i);
+                if (pgi.numProcs < min_procs) min_procs = pgi.numProcs;  //Finding the minimum number of processors in the Processor Groups
+                calculate_numa[i] = pgi.numProcs;
                 __TBB_ASSERT( pgi.numProcs <= (int)sizeof(DWORD_PTR) * CHAR_BIT, nullptr);
                 pgi.mask = pgi.numProcs == sizeof(DWORD_PTR) * CHAR_BIT ? ~(DWORD_PTR)0 : (DWORD_PTR(1) << pgi.numProcs) - 1;
                 pgi.numProcsRunningTotal = nprocs += pgi.numProcs;
             }
             __TBB_ASSERT( nprocs == (int)TBB_GetActiveProcessorCount( TBB_ALL_PROCESSOR_GROUPS ), nullptr);
+
+            for (WORD i = 0; i < ProcessorGroupInfo::NumGroups; ++i) {
+                if (i == 0) calculate_numa[i] = (calculate_numa[i] / min_procs);
+                else calculate_numa[i] = calculate_numa[i-1] + (calculate_numa[i] / min_procs);
+                
+            }
+
+            numaSum = calculate_numa[ProcessorGroupInfo::NumGroups - 1]+1;
```

Comment:
Why +1?  Maybe this is right, but it seems like you are calculating a prefix sum of the number of min_procs groups. I'm not sure why you are adjusting by 1.

Reply:
Should be +1 so that the ProcIdx in FindProcessorGroupIndex stays between 1 and calculate_numa[ProcessorGroupInfo::NumGroups - 1], but makes sense to have this later in the function. Fixed.